app/shared/components/Global/Form/Field/Token.js

```diff
@@ -1,36 +1,28 @@
 // @flow
 import React, { Component } from 'react';
-import { I18n } from 'react-i18next';
 import { Form, Input } from 'semantic-ui-react';
 
 import debounce from 'lodash/debounce';
 
 export default class FormFieldToken extends Component<Props> {
-  state = { value: '' };
   onChange = debounce((e, { name, value }) => {
-    const asset = 'EOS';
-    const parsed = (value > 0) ? `${value.toFixed(4)} ${asset}` : `0.0000 ${asset}`;
-    this.setState({
-      value: parsed,
-    }, () => {
-      this.props.onChange(e, { name, value: parsed });
-    });
+    this.props.onChange(e, { name, value });
```

Comment:
This shouldn't have been removed and migrated to the actions (`delegatebwParams ` and `undelegatebwParams`). 

The point was to have a global component that did the formatting for the application so we don't have to replicate the same code over and over in the actions for formatting token values.

Reply:
Hmm so you are talking about these two lines? ```const asset = 'EOS';
    const parsed = (value > 0) ? `${value.toFixed(4)} ${asset}` : `0.0000 ${asset}`;``` I removed that cause it seemed to apply more to a multi token form 😄 I also didn't like the idea of setting 0 for negative values as opposed to giving them an error. Maybe one thing we could do is move some of the logic from the `errorsInForm` method in Stake/Input to the TokenField component.